### PR TITLE
Add plugin support for streaming packets to Kafka

### PR DESCRIPTION
**Pull request by r8ygun**
There are some assumptions that I've made in this PR that probably need to be addressed before this can be marked as ready for review.

1. With the S3 Plugin, we added the PCAP header to each file so that the objects in S3 were valid files. It would seem sensible to do the same with Kafka but we might be more constrained.  For example, how do we know when to start a new "file"?  We probably don't want to add the header to every 65 KB message unless we also want files to be that size too, but that seems odd.  Maybe there should be some `fileSize` property and a new header is added to the first packet after `fileSize` amount of data has been received. There will be more work to do if this is the behaviour that you want.
2. Packets that are destined for the same file need to share a message key to guarantee ordering.  I've hard coded this as "packetstreamer" right now, but I'm happy to hear suggestions on other approaches. If we follow the approach suggestion in point 1, we could generate a new value for each "file" that could be used as the key.

### CODE REVIEW

File: contrib/config/receiver-kafka.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+input:
+  address: 0.0.0.0
+  port: 8081
+output:
+  file:
+    path: /dev/null
```
Comment (vadorovsky) [inline contrib/config/receiver-kafka.yaml:6]: Do we need to define the file section here at all?

File: pkg/plugins/kafka/kafka.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+				p.cleanup()
+				return
+			}
+		}
+	}()
+	return inputChan
+}
+
+func (p *Plugin) cleanup() {
+	if len(p.Buffer) > 4 {
```
Comment (vadorovsky) [inline pkg/plugins/kafka/kafka.go:98]: Why 4? Maybe we should have some constant with a comment for that?

**Review by vadorovsky — CHANGES_REQUESTED** [review]
Thanks a lot for working on that. Looking forward to see the final so look solution! 🙂

File: pkg/plugins/kafka/kafka.go (hunk 2 above)
Comment (r8ygun) [reply pkg/plugins/kafka/kafka.go:98]: This is very likely to change / be removed. The idea for 4 was to only flush if there's a buffer that contains more than the header length. But yeah, constant / comment will be necessary regardless so thank you.

File: contrib/config/receiver-kafka.yaml (hunk 1 above)
Comment (r8ygun) [reply contrib/config/receiver-kafka.yaml]: Removed.

**Comment by r8ygun** [comment]
There is now a `fileSize` property that is set in the config. We will read publish each "message" to Kafka, but after `fileSize` amount of data is received, we will start a new "file" (with accompanying header).  Each "file" has its own unique ID that is used as the Kafka message key when publishing packets that will be contained in that "file".

**Comment by vadorovsky** [comment]
Thanks, LGTM.

We need to do better job with handling errors, but it needs to be done for S3 as well, so I'd be happy to make that change for both plugins at the same time.

**Pull request merged**